File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaUserRoleRepository.java
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories;

import java.util.List;
import java.util.Set;
import java.util.UUID;

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.UserRoleEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface JpaUserRoleRepository extends JpaRepository<UserRoleEntity, UUID> {
    List<UserRoleEntity> findAllByUserId(UUID userId);

    UserRoleEntity findByUserIdAndRoleId(UUID userId, UUID roleId);

    List<UserRoleEntity> findAllByUserIdIn(Set<UUID> userIds);

    boolean existsByRoleIdAndDeletedAtIsNull(UUID roleId);

    List<UserRoleEntity> findAllByRoleId(UUID roleId);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaUserRoleRepository.kt
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.UserRoleEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface JpaUserRoleRepository : JpaRepository<UserRoleEntity, UUID> {
    fun findAllByUserId(userId: UUID): List<UserRoleEntity>
    fun findByUserIdAndRoleId(userId: UUID, roleId: UUID): UserRoleEntity?

    fun findAllByUserIdIn(userIds: Set<UUID>): List<UserRoleEntity>
}
